modules/analyzer.py

```diff
@@ -9,6 +9,13 @@
 from .getweblinks import get_urls_from_page
 from .pagereader import read
 
+
+class CustomTree(Tree): #override the Tree class to add an attribute which'll hold the page name
```

Comment:
Instead of creating a custom tree, this should be integrated into the `LinkTree`, through the style methods. We can extend the class if we see further need but I don't think it's necessary currently